cinema/views.py

```diff
@@ -21,32 +24,70 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    mixins.ListModelMixin, mixins.CreateModelMixin, viewsets.GenericViewSet
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
 
+    def get_permissions(self):
+        if self.action == "create":
+            self.permission_classes = [IsAdminUser]
+        else:
+            self.permission_classes = [IsAuthenticated]
+        return super().get_permissions()
```

Comment:
As I understand task, you already have implemented IsAdminORIfAuthenticatedReadOnly. That's the same what you wrote in def get_permissions. So like python dev - probably should just wrote permission_classes = (IsAdminOrIfAuthenticatedReadOnly,) and no need to write for every ViewSet +7 code lines, just add one attribute.